ACPT-674: ImportCsvApi Docs (#28)
AdobeDocs/commerce-webapi: src/pages/rest/modules/import/index.md

---
title: ImportCsv API
description: Import entities into Adobe Commerce using REST.
--- 

> 

Reviewer: @keharper, will you please add an H1 for this?

---
title: ImportCsv API

description: Import entities into Adobe Commerce using Rest API

--- 
## Import API

The  `StartImportInteface` service provides the means to efficiently import entities in Magento using a single API call. 

The `POST /rest/default/V1/import/csv` call allows import of the following entities:

> 

Reviewer 2: Use the entity names instead of the UI labels.

* Advanced Pricing
* Products (Simple & Configurable)
* Customers
* Customer Finances
* Customer Addresses

**Service Name:**

`StartImportInterface`

**REST Endpoint:**

```http

> POST /rest/default/V1/import/csv

Reviewer 2: Same comment about the route.

```

**StartImportInterface Parameters:**

|Name | Description  | Format | Requirements|
|-----|--------------|--------|---------------------------------------------------------------------------------|

> |`entity`|Type of entity to be imported| string | Required for determining type of entity being imported|

Reviewer 2: ```suggestion
|`entity` | The type of entity to be imported| string | Required. One of .... |
```

> |`behavior` | Import Behavior | string | Required for determining if entities are being added/updated,replaced or deleted |

Reviewer 2: ```suggestion
|`behavior` | The action  to perform | string | Required. One of .... |
```

> |`validationStrategy` | Strategy to use when entities are invalid | string | Required for determining validation strategy|

Reviewer 2: ```suggestion
|`validationStrategy` | Strategy to use when entities are invalid | string | Required. One of ....|
```

This needs to be discussed in the description of the endpoint.

> |`allowedErrorCount` | No. of errors before import is canceled | string | Required for canceling import after certain number of invalid rows|

Reviewer 2: ```suggestion
|`allowedErrorCount` | The maximum number of errors that can occur until import is canceled | string | Required |
```

What happens with the uploads that did not encounter errors before the error count is reached? Are they marked as successful, or are they rolled back?

> |`csvData` | Base64 encoded string containing csv data | string | Csv data is encoded in base64 format|

Reviewer 2: ```suggestion
|`csvData` | Base64 encoded string containing CSV data | string | Required |
```

**Sample Usage:**

`POST <host>/rest/default/V1/import/csv`

> <CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

Reviewer 2: Delete this line

**Headers:**

`Content-Type: application/json`

`Authorization: Bearer <administrator token>`

**Payload:**

```json
{
  "source":{
    "entity":"advanced_pricing", 
    "behavior":"append",
    "validationStrategy": "validation-stop-on-errors", 
    "allowedErrorCount":"10", 
    "csvData":"c2t1LHRpZXJfcHJpY2Vfd2Vic2l0ZSx0aWVyX3ByaWNlX2N1c3RvbWVyX2dyb3VwLHRpZXJfcHJpY2VfcXR5LHRpZXJfcHJpY2UsdGllcl9wcmljZV92YWx1ZV90eXBlClNpbXBsZTEsIkFsbCBXZWJzaXRlcyBbVVNEXSIsIk5PVCBMT0dHRUQgSU4iLDEuMDAwMCwyNTAuMDAwMDAwLEZpeGVk"
  }
}

```

> #### Response

Reviewer 2: ```suggestion
**Response:**
```
